File: VB/XtraReport1.vb
#Region "usings"
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraReports.Parameters
#End Region

Namespace WindowsFormsApp1
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport

		Public Sub New()
			InitializeComponent()
		End Sub

#Region "implementCustomParameterEditor"
		Private Sub XtraReport1_ParametersRequestBeforeShow(ByVal sender As Object, ByVal e As ParametersRequestEventArgs) Handles MyBase.ParametersRequestBeforeShow
			' Create and set up a radio group editor.
			Dim radioGroup = CreateRadioGroup()

			' Find a required parameter by name and assign
			' a custom editor to the Editor property.
			For Each info In e.ParametersInformation
				If info.Parameter.Name = "shape" Then
					info.Editor = radioGroup
					Exit For
				End If
			Next info
		End Sub

		Private Function CreateRadioGroup() As RadioGroup
			Dim radioGroup = New RadioGroup()

			Dim radioGroupItemTitles() As String = {"Circle", "Rectangle", "Ellipse", "Triangle", "Square"}

			For Each item In radioGroupItemTitles
				radioGroup.Properties.Items.Add(New RadioGroupItem(item, item))
			Next item

			radioGroup.Properties.ItemVertAlignment = RadioItemVertAlignment.Top
			radioGroup.MinimumSize = New System.Drawing.Size(0, 100)

			Return radioGroup
		End Function
#End Region
	End Class
End Namespace

File: CS/XtraReport1.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraReports.Parameters;

namespace WindowsFormsApp1 {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void XtraReport1_ParametersRequestBeforeShow(object sender, ParametersRequestEventArgs e) {
            // Create and set up a radio group editor.
            var radioGroup = CreateRadioGroup();

            // Find a required parameter by name and assign
            // a custom editor to the Editor property.
            foreach (var info in e.ParametersInformation) {
                if (info.Parameter.Name == "shape") {
                    info.Editor = radioGroup;
                    break;
                }
            }
        }

        private RadioGroup CreateRadioGroup() {
            var radioGroup = new RadioGroup();

            string[] radioGroupItemTitles = new string[] {
                "Circle", "Rectangle", "Ellipse",
                "Triangle", "Square"
            };

            foreach (var item in radioGroupItemTitles) {
                radioGroup.Properties.Items.Add(new RadioGroupItem(item, item));
            }

            radioGroup.Properties.ItemVertAlignment = RadioItemVertAlignment.Top;
            radioGroup.MinimumSize = new System.Drawing.Size(0, 100);

            return radioGroup;
        }
    }
}
